101-backup-create-disk-enable-protection/azuredeploy.json

```diff
@@ -0,0 +1,239 @@
+            "type": "[parameters('vaultStorageRedundancy')]"
+          }
+        ]
+      }
+    },
+    {
+      "type": "Microsoft.DataProtection/backupVaults/backupPolicies",
+      "apiVersion": "2020-01-01-alpha",
+      "name": "[concat(parameters('vaultName'), '/', parameters('backupPolicyName'))]",
+      "dependsOn": [
+        "[resourceId('Microsoft.DataProtection/backupVaults', parameters('vaultName'))]"
+      ],
+      "properties": {
+        "policyRules": [
+            {
+                "backupParameters": {
+                    "backupType": "Incremental",
+                    "objectType": "AzureBackupParams"
+                },
+                "trigger": {
+                    "schedule": {
+                        "repeatingTimeIntervals": [
+                            "[variables('repeatingTimeInterval')]"
+                        ],
+                        "timeZone": "UTC"
+                    },
+                    "taggingCriteria": [
+                        {
+                            "tagInfo": {
+                                "tagName": "Default",
+                                "id": "Default_"
+                            },
+                            "taggingPriority": 99,
+                            "isDefault": true
+                        }
+                    ],
+                    "objectType": "ScheduleBasedTriggerContext"
+                },
+                "dataStore": {
+                    "dataStoreType": "OperationalStore",
+                    "objectType": "DataStoreInfoBase"
+                },
+                "name": "BackupHourly",
+                "objectType": "AzureBackupRule"
+            },
+            {
+            "lifecycles": [
+              {
+                "sourceDataStore": {
+                  "dataStoreType": "OperationalStore",
+                  "objectType": "DataStoreInfoBase"
+                },
+                "deleteAfter": {
+                  "objectType": "AbsoluteDeleteOption",
+                  "duration": "[variables('retentionDuration')]"
+                },
+                "targetDataStoreCopySettings": []
+              }
+            ],
+            "isDefault": true,
+            "name": "Default",
+            "objectType": "AzureRetentionRule",
+            "ruleType": "Retention"
+          }
+        ],
+        "datasourceTypes": [
+          "[variables('dataSourceType')]"
+        ],
+        "objectType": "BackupPolicy"
+      }
+    },    
+    {
+      "type": "Microsoft.Compute/disks",
+      "apiVersion": "2020-12-01",
+      "name": "[parameters('diskName')]",
+      "location": "[parameters('location')]",
+      "properties": {
+        "creationData": {
+          "createOption": "Empty"
+        },
+        "diskSizeGB": 200
+      }
+    },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2020-04-01-preview",
+      "name": "[parameters('roleNameGuidForDisk')]",
```

Comment:
this must be deterministic and unique for the perms, role and scope (the parameter is not)... e.g.

```guid(resourceGroup().id, variables('roleDefinitionIdForDisk'), resourceId('Microsoft.DataProtection/backupVaults', parameters('vaultName'))```

Reply:
Addressed this.